giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/rest/FilesController.java

```diff
@@ -160,6 +164,88 @@
         return new ResponseEntity<String>(sw.toString(), HttpStatus.OK);
     }
 
+    @DocumentOwnerCheck
+    @RequestMapping(value = GET_DOCUMENT_PATH
+            + "/access/change", produces = "application/json;charset=UTF-8", method = RequestMethod.POST)
+    public ResponseEntity<String> changeDocumentAccess(@RequestParam(defaultValue = "") String accessToken,
+            HttpServletRequest request, @PathVariable("docId") String docId, @RequestParam("access") String access,
+            User user) {
+
+        if (docId == null || docId.isEmpty() || access == null || access.isEmpty()) {
+            return new ResponseEntity<String>("{'error': 'Missing document id or/and access type.'}",
+                    HttpStatus.BAD_REQUEST);
+        }
+        IDocument doc = filesManager.getDocument(docId);
+
+        if (doc == null) {
+            return new ResponseEntity<String>("{'error': 'Document does not exist.'}", HttpStatus.NOT_FOUND);
+        }
+
+        DocumentAccess docAccess = null;
+        try {
+            docAccess = DocumentAccess.valueOf(access);
+            if (docAccess == null) {
+                throw new IllegalArgumentException("Incorrect document access type");
+            }
```

Comment:
wait, why are you throwing an exception just to catch it right away? doesn't make sense.